src/ChangeStream.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/*
+ * Copyright 2017 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace MongoDB;
+
+use MongoDB\ChangeStreamIterator;
+use MongoDB\Driver\Manager;
+use MongoDB\Exception\ResumableException;
+use MongoDB\Operation\ChangeStreamCommand;
+
+/**
+ * Operation for the changeStream command.
+ *
+ * @api
+ * @see \MongoDB\Collection::changeStream()
+ * @see http://docs.mongodb.org/manual/reference/command/changeStream/
+ */
+class ChangeStream
+{
+    private $databaseName;
+    private $collectionName;
+    private $pipeline;
+    private $options;
+    private $resumeToken;
+    private $manager;
+    private $csIt;
+
+    public function __construct($cursor, $databaseName, $collectionName, array $pipeline, array $options = [], $resumeToken, Manager $manager)
+    {
+        $this->databaseName = (string) $databaseName;
+        $this->collectionName = (string) $collectionName;
+        $this->pipeline = $pipeline;
+        $this->options = $options;
+        $this->resumeToken = $resumeToken;
+        $this->manager = $manager;
+
+        $this->csIt = new ChangeStreamIterator($cursor);
+    }
+
+    public function current()
+    {
+        return $this->csIt->current();
+    }
+
+    public function getId()
+    {
+        return $this->csIt->getId();
```

Comment:
I had no idea what this was doing on the first run through, as ChangeStreamIterator has no `getId()` method. After reading the [IteratorIterator Notes](http://php.net/iteratoriterator), I realized it is calling `MongoDB\Driver\Cursor::getId()` due to the following behavior:

> This class [IteratorIterator] permits access to methods of the inner iterator via the __call magic method.

I think explicitly accessing the inner iterator and calling `getId()` would be preferable (for future maintainability and keeping the code readable).

Additionally, ChangeStreamIterator currently doesn't enforce the type of its inner iterator. In the interest of being defensive, it'd be wise to overload its constructor to type hint against the `MongoDB\Driver\Cursor` class before then calling `parent::__construct($cursor)`. This will ensure that it only ever wraps a cursor object.